index.js

```diff
@@ -110,6 +110,35 @@ if (argv.gasprice < 1 || argv.gasprice > 1000) throw "--gasprice must be between
   // Since the simulated transaction succeeded, send the real one to the network.
   const { transactionHash } = await perpetualCreator.methods.createPerpetual(perpetualParams, configSettings).send(transactionOptions);
   console.log("Deployed in transaction:", transactionHash);
+
+  // Log ABI encoded params for easier contract verification:
+  const newPerpetual = new web3.eth.Contract(
+    getAbi("Perpetual"),
+    address
+  ); 
+  let perpConstructorParams = {
+    ...perpetualParams,
+    configStoreAddress: await newPerpetual.methods.configStore().call(),
+    finderAddress: getAddress("Finder", networkId),
+    tokenFactoryAddress: getAddress("TokenFactory", networkId),
+    timerAddress: await perpetualCreator.methods.timerAddress().call(),
+    tokenAddress: await newPerpetual.methods.tokenCurrency().call()
+  };
+  const encodedParameters = web3.eth.abi.encodeParameters(getAbi("Perpetual", "latest")[0].inputs, [
+    perpConstructorParams
+  ]);
+  console.log("Encoded Perpetual Parameters", encodedParameters);
+  console.table(perpConstructorParams);
```

Comment:
nit: I think we should only print these if a user provides a flag. Most of the time, users won't care about this because etherscan will auto-verify all contracts after the first is verified.